controllers/secretgenerator/secretgenerator_controller.go

```diff
@@ -108,12 +108,14 @@ func (r *SecretGeneratorReconciler) Reconcile(ctx context.Context, request ctrl.
 	foundSecret := &corev1.Secret{}
 	err := r.Client.Get(ctx, request.NamespacedName, foundSecret)
 	if err != nil {
-		if k8serr.IsNotFound(err) {
-			// If Secret is deleted, delete OAuthClient if exists
-			err = r.deleteOAuthClient(ctx, request.Name)
+		if !k8serr.IsNotFound(err) {
+			return ctrl.Result{}, err
 		}
 
-		return ctrl.Result{}, err
+		// If Secret is deleted, delete OAuthClient if exists
+		if err = r.deleteOAuthClient(ctx, request.Name); err != nil {
+			return ctrl.Result{}, err
```

Comment:
A misleading error can also be reported here because by looking at the [deleteOAuthClient](https://github.com/opendatahub-io/opendatahub-operator/blob/b91bd2981c5fe5b9182fef006dffbab4d731e772/controllers/secretgenerator/secretgenerator_controller.go#L244-L259) function, it could be possible that the resource gets removed in between the `client.Get `and the `client.Delete` call, hence IMHO,  the `deleteOAuthClient` could avoid  performing a get+delete, but a delete + check for `NotFound`